jsonnet/components/prometheus-adapter-audit.libsonnet

```diff
@@ -0,0 +1,117 @@
+// NOTE: audit rules are copied from openshift/library-go/pkg/operator/apiserver/audit/manifests/*.yaml
+// to maintain the same UX as OpenShift apiserver
+
+// createProfile prepends base audit rules to the rules passed
+local createProfile(rules) =
+  // see openshift/library-go/pkg/operator/apiserver/audit/manifests/base-policy.yaml
+  local baseAuditPolicy = |||
+    apiVersion: audit.k8s.io/v1
+    kind: Policy
+    metadata:
+      name: policy
+    # Don't generate audit events for all requests in RequestReceived stage.
+    omitStages:
+    - "RequestReceived"
+    rules:
+    # Don't log requests for events
+    - level: None
+      resources:
+      - group: ""
+        resources: ["events"]
+    # Don't log authenticated requests to certain non-resource URL paths.
+    - level: None
+      userGroups: ["system:authenticated", "system:unauthenticated"]
+      nonResourceURLs:
+      - "/api*" # Wildcard matching.
+      - "/version"
+      - "/healthz"
+      - "/readyz"
+  |||;
+
+  |||
+    %s
+    %s
+  ||| % [baseAuditPolicy, rules];
+
+local defaultRules = |||
+  # Log the full Identity API resource object so that the audit trail
+  # allows us to match the username with the IDP identity.
+  - level: RequestResponse
+    verbs: ["create", "update", "patch", "delete"]
+    resources:
+    - group: "user.openshift.io"
+      resources: ["identities"]
+    - group: "oauth.openshift.io"
+      resources: ["oauthaccesstokens", "oauthauthorizetokens"]
+  # A catch-all rule to log all other requests at the Metadata level.
+  - level: Metadata
+    # Long-running requests like watches that fall under this rule will not
+    # generate an audit event in RequestReceived.
+    omitStages:
+    - "RequestReceived"
+|||;
+
+local allRequestBodiesRules = |||
+  # exclude resources where the body is security-sensitive
+  - level: Metadata
+    resources:
+    - group: "route.openshift.io"
+      resources: ["routes"]
+    - resources: ["secrets"]
+  - level: Metadata
+    resources:
+    - group: "oauth.openshift.io"
+      resources: ["oauthclients"]
+  # catch-all rule to log all other requests with request and response payloads
+  - level: RequestResponse
+|||;
+
+
+local writeRequestBodiesRules = |||
```

Comment:
@sthaha thinking about it again, this one doesn't make sense for prometheus-adapter since we won't receive any write requests on the metrics endpoint, only read ones. With that in mind, I am thinking that it might be better to stick to the upstream levels: https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#audit-policy in the end. wdyt?